### PR TITLE
Sd 1338 drop 1.4 api support

### PR DESCRIPTION
**Pull request by itorres**
Since the 1.4 API is not available anymore we should get rid of the legacy code supporting it.

### CODE REVIEW

**Comment by carlosperello** [comment]
It looks ok. I'm only missing the main README.md edit to remove references to the v1 API configuration keys.

File: README.md [hunk 1]
```diff
@@ -12,8 +12,6 @@ Puppet Module for deploying the Server Density Agent and agent plugins
 
 ### v2 API Config
```
Comment (dmytton) [inline README.md]: Can probably remove any reference to "v2" too

Comment (fullmooninu) [inline README.md]: https://github.com/serverdensity/puppet-serverdensity/blob/master/metadata.json#L28 bump this version requirement to 1.8.0 ?

**Comment by fullmooninu** [comment]
Tested and working

https://totoro.serverdensity.io/devices/55e3e75995fe35936e6773ae/overview  

*can be merged*

Remaining issues:
- here on git the module is named puppet-serverdensity yet inside the init it is named serverdensity_agent. The metadata is also fine, so I would consider renaming/moving this repository.
- on totoro, the device is still not listed as a working google device.

**Comment by carlosperello** [comment]
@fullmooninu please, could you file new JIRA reports for those remaining issues? This branch is just a cleanup branch so we shouldn't deal with any other issue this module may have.

**Pull request merged**